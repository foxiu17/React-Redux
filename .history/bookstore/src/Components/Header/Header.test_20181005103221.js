import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('AdminPanel tests: ', () => {
  it('AdminPanel renders without problems', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AdminPanel />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('AdminPanel hello world', () => {
    const wrapper = shallow(<AdminPanel />);
    expect(wrapper.find('h1').text()).toBe('Hello World');
  })
})