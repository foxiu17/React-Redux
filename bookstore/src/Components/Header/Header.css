.header {
  background-color: darkblue;
  color: aliceblue;
}