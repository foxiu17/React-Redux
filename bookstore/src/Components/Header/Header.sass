.header
  width: 100%
  background-color: rgb(99, 71, 30)
  color: aliceblue

.btn-goAdmin
  margin-right: 15px

