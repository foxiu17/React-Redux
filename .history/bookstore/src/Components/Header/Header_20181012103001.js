import React, { Component } from 'react';
import Link from 'react-router-dom';

import './Header.css';

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      bookstoreName: "Bad books",
      clicked: true,
      textColor: "red",
      backgroundColor: "black"
    }
  }

  handleClick = () => {
    if (this.state.clicked) {
      this.setState({
        bookstoreName: "Good books",
        textColor: "white",
        backgroundColor: "cyan",
        clicked: !this.state.clicked
      });
    } else {
      this.setState({
        textColor: "red",
        backgroundColor: "black",
        bookstoreName: "Bad books",
      });
    }
    this.setState({

      clicked: !this.state.clicked
    })
  }

  render() {
    let headerCss = {
      color: this.state.textColor,
      backgroundColor: this.state.backgroundColor
    }

    return (
      <div className="header" onClick={this.handleClick} style={headerCss}>
        <center><h1>{this.state.bookstoreName}</h1></center>
        <Link to=""
      </div>
    );
  }
}
export default Header;