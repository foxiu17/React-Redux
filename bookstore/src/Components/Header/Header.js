import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './Header.css';

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      bookstoreName: "Good bookstore",
      clicked: true,
      textColor: "black",
      backgroundColor: "rgb(99, 71, 30)"
    }
  }

  handleClick = () => {
    if (this.state.clicked) {
      this.setState({
        bookstoreName: "Good bookstore clicked",
        textColor: "white",
        backgroundColor: "cyan",
        clicked: !this.state.clicked
      });
    } else {
      this.setState({
        textColor: "black",
        backgroundColor: "rgb(99, 71, 30)",
        bookstoreName: "Good bookstore",
      });
    }
    this.setState({

      clicked: !this.state.clicked
    })
  }

  render() {
    let headerCss = {
      color: this.state.textColor,
      backgroundColor: this.state.backgroundColor
    }

    return (
      <div className="header" onClick={this.handleClick} style={headerCss}>
        <center><h1>{this.state.bookstoreName}</h1></center>
        <div className="row justify-content-end">
          <Link to="/admin" ><button className="btn btn-info btn-goAdmin">Administrator Panel</button></Link>
        </div>
      </div>
    );
  }
}
export default Header;