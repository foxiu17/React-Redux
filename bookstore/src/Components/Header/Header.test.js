import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Header tests: ', () => {
  it('Header renders without problems', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Header - title changes', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').text()).toBe('Bad books');
    wrapper.setState({
      bookstoreName : 'Good books'
    });
    expect(wrapper.find('h1').text()).toBe('Good books');
  });

  it('Header - changes after click', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.state().bookstoreName).toBe('Bad books');
    wrapper.find('.header').simulate('click');
    expect(wrapper.state().bookstoreName).toBe('Good books');
  });

  it('Snapshot matches', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  })
})