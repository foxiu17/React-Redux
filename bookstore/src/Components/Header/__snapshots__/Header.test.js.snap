// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests:  Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <center>
        <h1>
          Bad books
        </h1>
      </center>,
      "className": "header",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "black",
        "color": "red",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Bad books
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bad books",
        },
        "ref": null,
        "rendered": "Bad books",
        "type": "h1",
      },
      "type": "center",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <center>
          <h1>
            Bad books
          </h1>
        </center>,
        "className": "header",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "black",
          "color": "red",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Bad books
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bad books",
          },
          "ref": null,
          "rendered": "Bad books",
          "type": "h1",
        },
        "type": "center",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
